Tools/paletteable sprite editor/js/lib/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+					this.appiledkernelMatrix[x-this.radius][y-this.radius] += this.kernelMatrix[x-centerPixel_X][y-centerPixel_Y];
+				}
+			}
+		}
+	}
+	function AlphaBoundary(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+
+	AlphaBoundary.prototype.load = function(){
+		this.setAttribute("radius", 5);
+	};
+	
+	AlphaBoundary.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		var neighborhood = getAttribute("radius");
+		for(var y=0; y<imageOut.getHeight(); y++){
+			for(var x=0; x<imageOut.getWidth(); x++){
+				this.alphaRadius(imageOut, x, y, neighborhood);
+			}
+		}
+	};
+	
+	AlphaBoundary.prototype.alphaRadius = function(image, x, y, radius){
+		
+		var oldAlpha = image.getAlphaComponent(x, y);
+		var newAlpha;
+		var totalAlpha=0;
+		var totalPixels=0;
+		var hn = radius/2;
+		
+		for(var j=y-hn; j<y+hn; j++){
+			for(var i=x-hn; i<x+hn; i++){
+				
+				if(i >= 0 && i< image.getWidth() && j >= 0 && j < image.getHeight()){
+					totalAlpha += image.getAlphaComponent(i, j);
+					totalPixels++;
+				}
+			}
+		}
+		
+		newAlpha = totalAlpha/totalPixels;
+		
+		if(newAlpha < oldAlpha)
+		image.setAlphaComponent(x, y, newAlpha);
+	};
+
+
+	function AverageColor(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+
+	AverageColor.prototype.load = function(){
+	};
+	
+	AverageColor.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		
+		var totalR=0;
+		var totalG=0;
+		var totalB=0;
+		
+		for (var x = 0; x < imageIn.getWidth(); x++) {
+			for (var y = 0; y < imageIn.getHeight(); y++) {
+				totalR += imageIn.getIntComponent0(x,y);
+				totalG += imageIn.getIntComponent1(x,y);
+				totalB += imageIn.getIntComponent2(x,y);
+			}
+		}
+		
+		var totalPixels = imageIn.getWidth()*imageIn.getHeight();
+		totalR = Math.round(totalR/totalPixels);
+		totalG = Math.round(totalG/totalPixels);
+		totalB = Math.round(totalB/totalPixels);
+		
+		if(attributesOut != null){
+			attributesOut.set("averageColor", [totalR, totalG, totalB]);
+		}
+	};
+
+	function BlackAndWhite(){
+		MarvinAbstractImagePlugin.super(this);
+		this.MAX_RLEVEL = 0.03;
+		this.load();
+	}
+	
+	BlackAndWhite.prototype.load = function() {
+		this.grayScale = new GrayScale();
+		this.setAttribute("level", 10);
+	}
+	
+	BlackAndWhite.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		this.grayScale.process(imageIn, imageOut);
+		var level = this.getAttribute("level");
+		var rlevel = (level/100.0)*this.MAX_RLEVEL;
+		
+		var c=0;
+		var gray;
+		for(var y=0; y<imageOut.getHeight(); y++){
+			for(var x=0; x<imageOut.getWidth(); x++){
+				gray = imageIn.getIntComponent0(x, y);
+				
+				
+				if(gray <= 127){
+					gray = Math.max((gray * (1 - ((127-gray)*rlevel))),0);
+				}
+				else{
+					gray = Math.min(gray* (1+((gray-127)*rlevel)), 255);
+				}
+				
+				if(c++ < 1){
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)